### PR TITLE
[SPARK-9641] [DOCS] spark.shuffle.service.port is not documented

### PR DESCRIPTION
**Pull request by srowen**
Document spark.shuffle.service.{enabled,port}

CC @sryza @tgravescs
This is pretty minimal; is there more to say here about the service?

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -474,6 +474,22 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.shuffle.service.enabled</code></td>
+  <td>false</td>
+  <td>
+    Enables the external shuffle service. This service preserves the shuffle files written by 
+    executors so the executors can be safely removed. This must be enabled if
```
Comment (tgravescs) [inline docs/configuration.md]: we may want to add that the external shuffle service must be installed for this to work or point to docs describing that

Comment (srowen) [inline docs/configuration.md]: Pardon my ignorance -- i don't pay much attention to this part -- but what does this entail? like, is there a bit of doc I can point to here?

Comment (tgravescs) [inline docs/configuration.md]: http://spark.apache.org/docs/latest/job-scheduling.html#configuration-and-setup

So on YARN atleast, you have to install the shuffle jar into the YARN nodemanager as an auxiliary service. Then when you run your application on yarn you have to turn this config to true to tell it to use that service.  If you haven't installed it then turning this to true will cause the application to fail.

I haven't looked to see how this is installed in standalone (if its automatically there).

**Comment by tgravescs** [comment]
+1 lgtm

**Comment by andrewor14** [comment]
Ideally there should really be a section on the shuffle service itself outside of dynamic allocation. :) We still haven't got to that but I didn't realize we haven't even officially documented these two configs.

LGTM feel free to merge this.

**Pull request closed without merging**